src/sentry/static/sentry/app/views/performance/transactionDetails/quickTrace.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import {Location} from 'history';
+
+import Placeholder from 'app/components/placeholder';
+import {t, tn} from 'app/locale';
+import {OrganizationSummary} from 'app/types';
+import {Event} from 'app/types/event';
+import {getShortEventId} from 'app/utils/events';
+
+import QuickTraceQuery, {TraceLite} from './quickTraceQuery';
+import {EventNode, MetaData, NodesContainer} from './styles';
+import {isTransaction, parseTraceLite} from './utils';
+
+type Props = {
+  event: Event;
+  organization: OrganizationSummary;
+  location: Location;
+};
+
+export default function QuickTrace({event, organization, location}: Props) {
+  // non transaction events are currently unsupported
+  if (!isTransaction(event)) {
+    return null;
+  }
+
+  const traceId = event.contexts?.trace?.trace_id ?? '';
+
+  return (
+    <QuickTraceQuery event={event} location={location} orgSlug={organization.slug}>
+      {({isLoading, error, trace}) => {
+        const body = isLoading ? (
+          <Placeholder height="33px" />
+        ) : error || trace === null ? (
+          '\u2014'
+        ) : (
+          <QuickTraceLite event={event} trace={trace} />
+        );
```

Comment:
Looks like trace can be undefined too(no root) which isn't being caught here

Reply:
Good point. Going to coalesce it to null in `QuickTraceQuery` to make sure this doesn't happen. Should be fixed in 45c5ed3.